lib/mail_room/connection.rb

```diff
@@ -176,8 +176,14 @@ def new_message_ids
       # uid_search still leaves messages UNSEEN
       all_unread = imap.uid_search(@mailbox.search_command)
 
+      max_unread = @mailbox.limit_max_unread
```

Comment:
Could we just default to using `@mailbox.limit_max_unread` if it isn't `nil`? Always do the `slice()`?

That would just leave some unread for the next loop, yeah?

Reply:
sure, assumably then logging a warning also goes away

Reply:
Yeah, I think no warning is fine. It just feels like we need to "paginate" over the unreads.

Is there some relatively high default limit that mostly doesn't cause the error? Or do we at what length the error typically shows up?

Reply:
Im guessing a limit of 32k should be fine.... but thats why I explicitly left it "null" so those running into this problem can change it to suite their specific needs.... then again, loading 32k emails might consume quite a lot of memory...

If I had more time, Id really like to rework the logic in such a way that it read batches of say 100 and does not idle unless theres nothing more to read

Reply:
It's been awhile since I looked at in detail, but IIRC, that's what the loop does now. It doesn't idle unless there are no messages. Just the "batch" size is as big as we can get (or too big as this error is saying).